src/ragger/backend/speculos.py

```diff
@@ -85,18 +86,21 @@ def send_raw(self, data: bytes = b"") -> None:
         logger.debug("Sending '%s'", data.hex())
         self._pending = ApduResponse(self._client._apdu_exchange_nowait(data))
 
-    @manage_error
+    @raise_policy_enforcer
     def receive(self) -> RAPDU:
         assert self._pending is not None
         result = RAPDU(0x9000, self._pending.receive())
-        logger.debug("Receiving '%s'", result)
         return result
 
-    @manage_error
+    @raise_policy_enforcer
     def exchange_raw(self, data: bytes = b"") -> RAPDU:
         logger.debug("Sending '%s'", data.hex())
         return RAPDU(0x9000, self._client._apdu_exchange(data))
 
+    @raise_policy_enforcer
+    def get_last_async_response(self, response) -> RAPDU:
```

Comment:
I'm not sure this method should be "public", it seems to be purely internal to this class. If it is (internal), prefix it with a `_` to explicit this should not be used from outside.